### PR TITLE
Exemplars: Update UX to match new tooltips

### LINKED ISSUE #79717: Exemplars: Match UI to tooltips
Design

Screenshot 2023-12-20 at 12 00 27 PM

### PR DESCRIPTION
**Pull request by adela-almasan**
https://github.com/grafana/grafana/assets/88068998/ba285d6a-b4c0-4004-99bf-139eb31c5908

Under `newVizTooltips` feature flag.

Fixes #79717 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/visualization/data-hover/DataHoverView.tsx [hunk 1]
```diff
@@ -110,6 +113,10 @@ export const DataHoverView = ({ data, rowIndex, columnIndex, sortOrder, mode, he
 
   const { displayValues, links } = dispValuesAndLinks;
 
+  if (config.featureToggles.newVizTooltips) {
+    return <ExemplarHoverView displayValues={displayValues} links={links} header={header} />;
+  }
```
Comment (nmarrs) [inline public/app/features/visualization/data-hover/DataHoverView.tsx:118]: I think we use DataHoverView in a few other places (outside of the exemplar use case) 🤔

See this example with BarChart
needs to be exemplar specific


I think given this we might want to have some way of determining whether or not we are displaying an exemplar - maybe for time being if the passed header prop = `Exemplar` or something (not ideal but 🤷🏻‍♂️)

**Review by nmarrs — COMMENTED** [review]
Code / style wise I think this is looking good - I think we do need to address the fact that `DataHoverView` is being used in multiple different places (and thus impacts other panels / areas of code for time being 😬)

File: public/app/features/visualization/data-hover/DataHoverView.tsx (hunk 1 above)
Comment (adela-almasan) [reply public/app/features/visualization/data-hover/DataHoverView.tsx:118]: Totally missed that. I added a check for the header.

**Review by nmarrs — APPROVED** [review]
🚀

**Pull request merged**